lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -447,3 +469,274 @@ def determine_optimal_points(frame_idx, coordinate, max_iterations=100):
+    data was introduced in [12]_.
+
+    Note that this estimator is only valid in the case of pure diffusion without drift.
+
+    Parameters
+    ----------
+    frame_indices : array_like
+        Frame indices
+    x : array_like
+        Positions over time
+    dt : float
+        Time step
+    blur_constant : float
+        Motion blur coefficient. Should be between 0 and 1/4.
+
+        The normalized shutter function is defined as :math:`c(t)`, where :math:`c(t)` represents
+        whether the shutter is open or closed. :math:`c(t)` is normalized w.r.t. area. For no
+        motion blur, :math:`c(t) = \delta(t_{exposure})`, whereas for a constantly open shutter it is
+        defined as :math:`c(t) = 1 / \Delta t`.
+
+        With :math:`c(t)` defined as before, the motion blur constant is defined as:
+
+        .. math::
+
+            R = \frac{1}{\Delta t} \int_{0}^{\Delta t}S(t) \left(1 - S(t)\right)dt
+
+        with
+
+        .. math::
+
+            S(t) = \int_{0}^{t} c(t') dt'
+
+        When there is no motion blur, we obtain: R = 0, whereas a continuously open shutter over the
+        exposure time results in R = 1/6. Note that when estimating both localization uncertainty
+        and the diffusion constant, the motion blur factor has no effect on the estimate of the
+        diffusion constant itself, but it does affect the calculated uncertainties. In the case of
+        a provided localization uncertainty, it does impact the estimate of the diffusion constant.
+    variance_loc : float
+        Variance of the localization error if it has been determined independently.
+    variance_variance_loc : float
+        The variance of the localization variance estimate (needed if variance_loc is provided).
+
+    Note
+    ----
+    With the SNR defined as:
+
+    .. math::
+
+        \sqrt{D \Delta t} / \sigma
+
+    With :math:`D` the diffusion constant and :math:`\sigma`
+    This method requires SNR > 1. Below SNR = 1 it degrades rapidly [11]_.
+
+    References
+    ----------
+    .. [11] Vestergaard, C. L., Blainey, P. C., & Flyvbjerg, H. (2014). Optimal estimation of
+            diffusion coefficients from single-particle trajectories. Physical Review E, 89(2),
+            022726.
+    .. [12] Vestergaard, C. L. (2016). Optimizing experimental parameters for tracking of diffusing
+            particles. Physical Review E, 94(2), 022401.
+    """
+    if not 0 <= blur_constant <= 0.25:
+        raise ValueError("Motion blur constant should be between 0 and 1/4")
+
+    if len(x) < 3:
+        raise RuntimeError("Insufficient intervals for using the CVE")
+
+    # Determine the average step size
+    average_frame_step = np.mean(np.diff(frame_indices))
+    avg_dt = average_frame_step * dt
+
+    dx = np.diff(x)
+    mean_dx_squared = np.mean(dx**2)
+
+    if not variance_loc:
+        # Estimate the variance based on this frame
```

Comment:
Is it just one frame? I thought x (and therefore dx) are determined from many consecutive frames?

Reply:
Good catch! Clearly that should've been `track`. Fixing.